pkg/cloudprovider/fake/factory.go

```diff
@@ -43,10 +45,14 @@ func NewNotImplementedFactory() *Factory {
 }
 
 func (f *Factory) NodeGroupFor(sng *v1alpha1.ScalableNodeGroupSpec) cloudprovider.NodeGroup {
-	return &NodeGroup{
-		WantErr:  f.WantErr,
-		Replicas: ptr.Int32(f.NodeReplicas[sng.ID]),
+	if f.nodegroup == nil {
```

Comment:
I'm not sure of the benefit here... Looks like you're creating a singleton pattern for nodegroup such that we always return the same object?

Reply:
Yes, trying to reuse the same nodegroup object because when `controller.reconcile` calls `c.CloudProvider.NodeGroupFor` it returns a new nodegroup object [here](https://github.com/awslabs/karpenter/blob/main/pkg/controllers/scalablenodegroup/v1alpha1/controller.go#L49) which cannot be referenced in tests and can't verify if the replicas set are correct since the object changes.